Add a section on keeping things up to date in the git section (#1031)
rust-lang/rustc-dev-guide: src/contributing.md

GitHub itself has some [great documentation][about-pull-requests] on using the
Pull Request feature. We use the "fork and pull" model [described here][development-models],
where contributors push changes to their personal fork and create pull requests to
bring those changes into the source repository. We have more info about how to use git

> in the `rust-lang` org under [the git section](./git.md).

Reviewer: ```suggestion
when contributing to Rust under [the git section](./git.md).
```
